Migrate from JUnit4 to JUnit5

rmcdouga opened the pull request:
I migrated to JUnit5 and updated a couple of other dependencies to the latest versions.

Resolves #1

rmcdouga committed:
Create branch as prep for migration to JUnit5.

rmcdouga committed:
Migrated from JUnit4 to JUnit5.  Updated some of the other dependencies to the latest versions.

dirtylabcoat reviewed (changes requested):
In order for the JUnit5-tests to run version 2.22+ of the Surefire plugin needs to be added to the plugin section of the generated pom.xml.
Likes this:
```
<plugin>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.22.2</version>
            </plugin>
```

rmcdouga committed:
Updated to use later release of sure-fire plugin.

rmcdouga commented:
Sorry - I updated the project, had it running in Eclipse but didn't run it through Maven.  My bad.  Thanks for diagnosing the problem and providing a solution.  I've updated the PR.  Sorry for the trouble.

dirtylabcoat reviewed (approved):
Nice work.
